### PR TITLE
Update truffle-artifactor (docs & codebase)

### LINKED ISSUE #1096: truffle-artifactor README.md is out of date
As pointed out on this old issue: https://github.com/trufflesuite/truffle-artifactor/issues/49  the README.md file needs to be updated accordingly.

### PR DESCRIPTION
**Pull request by CruzMolina**
This PR updates the documentation to `truffle-artifactor`'s current behavior & API. Also burns some code bringing the package up to ES6, converts the API into async methods, & intends to make the code easier to reason while debugging.

Resolves #1096.

### CODE REVIEW

File: packages/truffle-artifactor/README.md [hunk 1]
```diff
@@ -43,30 +31,31 @@ $ npm install truffle-artifactor
...
-});
+artifactor.save(contract_data);
+// The file ./MyContract.json now exists, which you can
+// import into your project like any other Javascript file.
 ```
+
 # API
 
-#### `artifactor.save(options, filename[, extra_options])`
+#### `artifactor.save(contract_data)`
```
Comment (gnidan) [inline packages/truffle-artifactor/README.md]: Might as well start using camelCase here

Comment (gnidan) [reply packages/truffle-artifactor/README.md]: Oh btw! We should update the code examples to use current schema. It's now `contractName`, and the like.

File: packages/truffle-artifactor/index.js [hunk 2]
```diff
@@ -4,97 +4,76 @@ var path = require("path");
...
-        // normalize existing and merge into final
-        finalObject = Schema.normalize(existingObjDirty);
-
-        // merge networks
-        var finalNetworks = {};
-        _.merge(finalNetworks, finalObject.networks, object.networks);
-
-        // update existing with new
-        _.assign(finalObject, object);
-        finalObject.networks = finalNetworks;
```
Comment (gnidan) [inline packages/truffle-artifactor/index.js]: Careful! I don't think we can delete this separate step to do the network merging.

File: packages/truffle-core/lib/testing/soliditytest.js [hunk 3]
```diff
@@ -152,7 +152,7 @@ const SolidityTest = {
           });
 
           runner.config.artifactor
-            .saveAll(contracts, runner.config.contracts_build_directory)
```
[the remark this answers is not in the document]
Comment (CruzMolina) [reply packages/truffle-core/lib/testing/soliditytest.js]: `saveAll()` & `save()` no longer have a destination parameter.

File: packages/truffle-artifactor/index.js (hunk 2 above)
Comment (CruzMolina) [reply packages/truffle-artifactor/index.js]: I _think_ I verified this worked fine locally (re-migrating on the same network & also re-migrating onto a diff chain), but sounds like a good candidate for a test.

File: packages/truffle-core/lib/testing/soliditytest.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (gnidan) [reply packages/truffle-core/lib/testing/soliditytest.js]: oh so they don't. and didn't before this PR. a relic!

File: packages/truffle-artifactor/index.js (hunk 2 above)
Comment (gnidan) [reply packages/truffle-artifactor/index.js]: good thinking!

File: packages/truffle-artifactor/index.js [hunk 4]
```diff
@@ -4,97 +4,76 @@ var path = require("path");
...
+    const normalizedArtifact = Schema.normalize(artifactObject);
+    const contractName = normalizedArtifact.contractName;
+
+    if (!contractName) throw new Error("You must specify a contract name.");
+
+    const output_path = path.join(self.destination, `${contractName}.json`);
+    let completeArtifact = {};
+
+    // helper for writing artifacts
+    const writeArtifact = _completeArtifact => {
```
[the remark this answers is not in the document]
Comment (CruzMolina) [reply packages/truffle-artifactor/index.js]: I _think_ this is already a private method.

File: packages/truffle-artifactor/index.js [hunk 5]
```diff
@@ -4,97 +4,76 @@ var path = require("path");
...
+
+    const normalizedArtifact = Schema.normalize(artifactObject);
+    const contractName = normalizedArtifact.contractName;
+
+    if (!contractName) throw new Error("You must specify a contract name.");
+
+    const output_path = path.join(self.destination, `${contractName}.json`);
+    let completeArtifact = {};
+
+    // helper for writing artifacts
```
Comment (CruzMolina) [inline packages/truffle-artifactor/index.js]:
```suggestion
    // private helper for writing artifacts
```

File: packages/truffle-artifactor/index.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (CruzMolina) [reply packages/truffle-artifactor/index.js]: documented 🎉

File: packages/truffle-artifactor/index.js [hunk 6]
```diff
@@ -4,97 +4,76 @@ var path = require("path");
...
-    object = Schema.normalize(object);
+class Artifactor {
+  constructor(destination) {
+    this.destination = destination;
+  }
 
-    if (object.contractName == null) {
-      return reject(new Error("You must specify a contract name."));
+  async save(artifactObject) {
+    const self = this;
```
Comment (eggplantzzz) [inline packages/truffle-artifactor/index.js]: Looks like you can get rid of this garbage and just use `this`.

File: packages/truffle-artifactor/index.js [hunk 7]
```diff
@@ -4,97 +4,76 @@ var path = require("path");
...
+
+    const normalizedArtifact = Schema.normalize(artifactObject);
+    const contractName = normalizedArtifact.contractName;
+
+    if (!contractName) throw new Error("You must specify a contract name.");
+
+    const output_path = path.join(self.destination, `${contractName}.json`);
```
Comment (eggplantzzz) [inline packages/truffle-artifactor/index.js]: Yeah and this can be `this.destination`.

File: packages/truffle-artifactor/index.js [hunk 8]
```diff
@@ -4,97 +4,76 @@ var path = require("path");
...
-        // merge networks
-        var finalNetworks = {};
-        _.merge(finalNetworks, finalObject.networks, object.networks);
-
-        // update existing with new
-        _.assign(finalObject, object);
-        finalObject.networks = finalNetworks;
-      }
+  async saveAll(artifactObjects) {
+    const self = this;
```
Comment (eggplantzzz) [inline packages/truffle-artifactor/index.js]: Same for this function.

File: packages/truffle-artifactor/test/contracts.js [hunk 9]
```diff
@@ -149,28 +147,26 @@ describe("artifactor + require", function() {
...
-      })
-      .then(function(value) {
+      .then(() => example.value.call())
+      .then(value => {
         assert.equal(parseInt(value), 25, "Ending value should be twenty-five");
         // BigNumber passed in a call.
         return example.parrot.call(865);
       })
-      .then(function(parrot_value) {
+      .then(parrot_value => {
```
Comment (princesinha19) [inline packages/truffle-artifactor/test/contracts.js]: same here. I would change is to parrotValue.

File: packages/truffle-artifactor/test/contracts.js [hunk 10]
```diff
@@ -51,94 +49,94 @@ describe("artifactor + require", function() {
...
+        const json = requireNoCache(expected_filepath);
         Example = contract(json);
         Example.setProvider(provider);
       });
   });
 
-  before(function() {
-    return web3.eth.getAccounts().then(accs => {
+  before(() =>
+    web3.eth.getAccounts().then(accs => {
```
Comment (princesinha19) [inline packages/truffle-artifactor/test/contracts.js]: I think we shouldn't use abbreviations. I would like to change **accs** to **accounts**.

**Comment by CruzMolina** [comment]
Good catches @princesinha19. I didn't scrutinize the tests as much, but since this isn't merged yet, I'll push the requested changes.

**Pull request merged**